python/ray/tune/tests/test_sample.py

```diff
@@ -1038,12 +1055,41 @@ def testConvertOptuna(self):
             metric="a",
             mode="max")
 
+        sampler4 = RandomSampler(seed=1234)
+        searcher4 = OptunaSearch(
+            space=optuna_define_by_run,
+            sampler=sampler4,
+            metric="a",
+            mode="max")
+
+        sampler5 = RandomSampler(seed=1234)
+        searcher5 = OptunaSearch(
+            space=optuna_define_by_run_with_constants,
+            sampler=sampler5,
+            metric="a",
+            mode="max")
+
+        config_constant = searcher5.suggest("0")
+        self.assertIn("constant", config_constant)
+
+        sampler6 = RandomSampler(seed=1234)
+        searcher6 = OptunaSearch(
+            space=optuna_define_by_run_invalid,
+            sampler=sampler6,
+            metric="a",
+            mode="max")
+
+        with self.assertRaises(TypeError):
+            searcher6.suggest("0")
+
         config1 = searcher1.suggest("0")
         config2 = searcher2.suggest("0")
         config3 = searcher3.suggest("0")
+        config4 = searcher4.suggest("0")
 
         self.assertEqual(config1, config2)
         self.assertEqual(config1, config3)
+        self.assertEqual(config1, config4)
```

Comment:
Shouldn't we have a test for `config_constant` here as well? (we can pop the constant key)